openstudiocore/src/model/ThermalZone.cpp

```diff
@@ -998,14 +993,32 @@ namespace detail {
 
   bool ThermalZone_Impl::setThermostat(const Thermostat & thermostat)
   {
-    resetThermostat();
-    return setPointer(OS_ThermalZoneFields::ThermostatName, thermostat.handle());
+    auto t_model = model();
+
+    if( t_model != thermostat.model() ) {
+      return false;
+    }
+
+    if( auto currentZone = thermostat.thermalZone() ) {
+      if( currentZone->handle() == handle() ) {
+        // or should it be false?
+        // I think this is similar to what you would see in 
+        // Lights::setSpace() under similar conditions
```

Comment:
True this can happen in the setSpace but there setting the same thing twice is benign, except I guess you get unnecessary change signals emitted